File: eDentist/UI/edentist_mobile/lib/models/comment.dart
import 'package:json_annotation/json_annotation.dart';

part "comment.g.dart";

@JsonSerializable()
class Comment {
  int? commentID;
  int? userId;
  DateTime? datum;
  int? blogId;
  String? content;

  Comment(this.commentID, this.userId, this.datum,
      this.blogId,this.content);

  factory Comment.fromJson(Map<String, dynamic> json) =>
      _$CommentFromJson(json);
  Map<String, dynamic> toJson() => _$CommentToJson(this);
}

File: eDentist/eDentist.Services/Database/Comment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eDentist.Services.Database
{
    public class Comment
    {
        [Key]
        public int CommentID { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; }
        public int UserId { get; set; }
        public DateTime Datum { get; set; }
        [ForeignKey("BlogId")]
        public BlogPost Blog { get; set; }
        public int BlogId { get; set; }

        public string Content { get; set; }
    }
}
